manager/dispatcher/dispatcher_test.go

```diff
@@ -1141,9 +1223,18 @@ func makeTasksAndDependencies(t *testing.T, nodeID string) ([]*api.Secret, []*ap
 				Data: []byte(fmt.Sprintf("config%d", i)),
 			},
 		})
+
+		resourceRefs = append(resourceRefs, &api.ResourceReference{
+			ResourceID:   fmt.Sprintf("IDresourceRefSecret%d", i),
+			ResourceType: api.ResourceType_SECRET,
+		}, &api.ResourceReference{
+			ResourceID:   fmt.Sprintf("IDresourceRefConfig%d", i),
+			ResourceType: api.ResourceType_CONFIG,
+		})
 	}
+
 	for i, taskState := range taskStatesInOrder {
-		spec := taskSpecFromDependencies(secrets[i], secrets[len(secrets)-1], configs[i], configs[len(configs)-1])
+		spec := taskSpecFromDependencies(secrets[i], secrets[len(secrets)-1], configs[i], configs[len(configs)-1], resourceRefs[2*i], resourceRefs[2*i+1])
```

Comment:
Let's make sure to test the following cases:

- Task with only a `SecretReference` and or `ConfigReference` (this is the case we need to handle for backward compatibility)
- Task with a `SecretReference` or `ConfigReference` that points to the same resource as a `ResourceReference` (this will be the new approach going forward, separating dependency linkage from runtime-specific configuration)
- Task with only a `ResourceReference` (corresponds to something like a `PluginSpec` that swarmkit doesn't directly understand)